src/irmin-pack/irmin_pack_intf.ml

```diff
@@ -35,6 +35,7 @@ module type Sigs = sig
     ?index_log_size:int ->
     ?merge_throttle:Conf.merge_throttle ->
     ?freeze_throttle:Conf.freeze_throttle ->
+    ?indexing_strategy:Pack_store.Indexing_strategy.t ->
```

Comment:
Don't know what you will think of this but... with lots of optional arguments like this, perhaps it is cleaner to have a record type `{ fresh:bool; readonly:bool; ...}` and a default value for the type, and let the user override values in the default if they want something different from the default. Not sure.. just a suggestion based on the observation that these long lists of optional args tend to infect the codebase and are somewhat verbose.

Reply:
I agree that having a concrete internal config type sounds like a good idea, to avoid some proliferation of these arguments in e.g. `ext.ml`, but I think it'd be better not to surface that to the user: a single constructor with optional arguments is easy to convert to a record internally and we retain an abstraction barrier.